### PR TITLE
ghost-cli: init at 1.26.1

### PR DESCRIPTION
**Pull request by cything**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by DimitarNestorov — APPROVED** [review]
You can also add a version test:

```nix
  passthru.tests = {
    version = testers.testVersion {
      package = ghost-cli;
    };
  };
```

LGTM

<details>
<summary>Manual tests</summary>


```sh
[nix-shell:~/.cache/nixpkgs-review/pr-366378]$ ghost
No command specified. Run `ghost help` for usage

[nix-shell:~/.cache/nixpkgs-review/pr-366378]$ ghost --version

Love open source? We’re hiring JavaScript Engineers to work on Ghost full-time.
https://careers.ghost.org


Ghost-CLI version: 1.26.1

[nix-shell:~/.cache/nixpkgs-review/pr-366378]$ ghost help
ghost [command]

Commands:
  ghost backup                 Backup content & files
  ghost buster                 Who ya gonna call? (Runs `yarn cache clean`)
  ghost check-update           Check if an update is available for a Ghost installation
  ghost config [key] [value]   View or edit Ghost configuration
  ghost doctor [categories..]  Check the system for any potential hiccups when installing/updating Ghost
  ghost export file            Export content from a blog
  ghost import [file]          Import a Ghost export
  ghost install [version]      Install a brand new instance of Ghost
  ghost log [name]             View the logs of a Ghost instance
  ghost ls                     View running ghost processes
  ghost migrate                Run system migrations on a Ghost instance
  ghost restart                Restart the Ghost instance
  ghost run                    Run a Ghost instance directly (used by process managers and for debugging)
  ghost setup [stages..]       Setup an installation of Ghost (after it is installed)
  ghost start [name]           Start an instance of Ghost
  ghost stop [name]            Stops an instance of Ghost
  ghost uninstall              Remove a Ghost instance and any related configuration files
  ghost update [version]       Update a Ghost instance
  ghost version                Prints out Ghost-CLI version (and Ghost version if one exists)

Global Options:
      --help         Show help                                                                            [boolean]
  -d, --dir          Folder to run command in
  -D, --development  Run in development mode                                                              [boolean]
  -V, --verbose      Enable verbose output                                                                [boolean]
      --prompt       [--no-prompt] Allow/Disallow UI prompting                            [boolean] [default: true]
      --color        [--no-color] Allow/Disallow colorful logging                         [boolean] [default: true]
      --auto         Automatically run as much as possible                               [boolean] [default: false]

For more information, see our docs at https://ghost.org/docs/ghost-cli/
```

</details>


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 366378`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ghost-cli</li>
  </ul>
</details>

**Comment by cything** [comment]
I used `versionCheckHook` instead cause the docs seemed to [prefer that](https://nixos.org/manual/nixpkgs/stable/#tester-testVersion). Also rebased it on top of `master`.

File: pkgs/by-name/gh/ghost-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  yarnOfflineCache = fetchYarnDeps {
+    yarnLock = finalAttrs.src + "/yarn.lock";
+    hash = "sha256-No+Hkb2ivrCSd0S9L5QxZ8ReX9NANMRITKHFvjzRSuc=";
+  };
+
+  nativeBuildInputs = [
+    yarnConfigHook
+    yarnInstallHook
+    versionCheckHook
```
Comment (DimitarNestorov) [inline pkgs/by-name/gh/ghost-cli/package.nix:30]: Put this under `nativeInstallCheckInputs`

**Review by DimitarNestorov — CHANGES_REQUESTED** [review]

**Review by DimitarNestorov — APPROVED** [review]

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

File: pkgs/by-name/gh/ghost-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ghost-cli";
+  version = "1.26.1";
+
+  src = fetchFromGitHub {
+    owner = "TryGhost";
+    repo = "Ghost-CLI";
+    rev = "v${finalAttrs.version}";
```
[the remark this answers is not in the document]
Comment (cything) [reply pkgs/by-name/gh/ghost-cli/package.nix]: Done

File: pkgs/by-name/gh/ghost-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  ];
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  doInstallCheck = true;
+  versionCheckProgram = ''${placeholder "out"}/bin/ghost'';
+
+  meta = {
+    description = "CLI Tool for installing & updating Ghost";
+    mainProgram = "ghost";
+    homepage = "https://ghost.org/docs/ghost-cli/";
+    license = lib.licenses.mit;
```
[the remark this answers is not in the document]
Comment (cything) [reply pkgs/by-name/gh/ghost-cli/package.nix]: Done

File: pkgs/by-name/gh/ghost-cli/package.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  nativeBuildInputs = [
+    yarnConfigHook
+    yarnInstallHook
+  ];
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  doInstallCheck = true;
+  versionCheckProgram = ''${placeholder "out"}/bin/ghost'';
+
+  meta = {
```
[the remark this answers is not in the document]
Comment (cything) [reply pkgs/by-name/gh/ghost-cli/package.nix]: Done

**Review by GaetanLepage — APPROVED** [review]
LGTM, thanks !

**Comment by cything** [comment]
> LGTM, thanks !

Thanks for the review and suggestions!

**Pull request merged**